libraries/Microsoft.Bot.Builder.Dialogs.Rules/Rules/UtteranceRecognizeRule.cs

```diff
@@ -35,19 +35,47 @@ protected override async Task<bool> OnIsTriggeredAsync(PlanningContext planning,
                     return false;
                 }
 
-                if (!recognizerResult.Intents.Any(r => r.Key == Intent))
+                if(recognizerResult.GetTopScoringIntent().score < 0.5)
```

Comment:
As we discussed, this is because of the luisRecognizer always get all the Intents with scores, that's why we need a threshold here.

---
In reply to: [266529972](https://github.com/Microsoft/botbuilder-dotnet/pull/1462#discussion_r266529972) [](ancestors = 266529972)